astronomer/providers/snowflake/hooks/sql_api_generate_jwt.py

```diff
@@ -0,0 +1,167 @@
+            # Handle the general case.
+            idx = account.find(".")
+            if idx > 0:
+                account = account[0:idx]
+        else:
+            # Handle the replication case.
+            idx = account.find("-")
+            if idx > 0:
+                account = account[0:idx]  # pragma: no cover
+        # Use uppercase for the account identifier.
+        return account.upper()
+
+    def get_token(self) -> Optional[Text]:
+        """
+        Generates a new JWT. If a JWT has been already been generated earlier, return the previously
+        generated token unless the specified renewal time has passed.
+        """
+        now = datetime.now(timezone.utc)  # Fetch the current time
+
+        # If the token has expired or doesn't exist, regenerate the token.
+        if self.token is None or self.renew_time <= now:
+            logger.info(
+                "Generating a new token because the present time (%s) is later than the renewal time (%s)",
+                now,
+                self.renew_time,
+            )
+            # Calculate the next time we need to renew the token.
+            self.renew_time = now + self.renewal_delay
+
+            # Prepare the fields for the payload.
+            # Generate the public key fingerprint for the issuer in the payload.
+            public_key_fp = self.calculate_public_key_fingerprint(self.private_key)
+
+            # Create our payload
+            payload = {
+                # Set the issuer to the fully qualified username concatenated with the public key fingerprint.
+                ISSUER: self.qualified_username + "." + public_key_fp,
+                # Set the subject to the fully qualified username.
+                SUBJECT: self.qualified_username,
+                # Set the issue time to now.
+                ISSUE_TIME: now,
+                # Set the expiration time, based on the lifetime specified for this object.
+                EXPIRE_TIME: now + self.lifetime,
+            }
+
+            # Regenerate the actual token
+            token = jwt.encode(payload, key=self.private_key, algorithm=JWTGenerator.ALGORITHM)
+            # If you are using a version of PyJWT prior to 2.0, jwt.encode returns a byte string, rather than a string.
+            # If the token is a byte string, convert it to a string.
+            if isinstance(token, bytes):
+                token = token.decode("utf-8")
+            self.token = token
+            logger.info(
+                "Generated a JWT with the following payload: %s",
+                jwt.decode(
+                    self.token, key=self.private_key.public_key(), algorithms=[JWTGenerator.ALGORITHM]
+                ),
+            )
+
+        return self.token
+
+    def calculate_public_key_fingerprint(self, private_key: Any) -> Text:
+        """
+        Given a private key in PEM format, return the public key fingerprint.
+        :param private_key: private key
```

Comment:
there should be an empty line between parm and description

Reply:
Done